### PR TITLE
bug: Fix #6439 issue in `StaticLambda` fixer

### LINKED ISSUE #6439: Bug in `static_lambda` rule
Hi,

It looks like the `static_lambda` rule is not working correctly in this case. Even if the case is totally unlikely to happen, it should not add the `static` keyword in the front.

```php
<?php

declare(strict_types=1);

namespace App;

class Foo
{
    public function foo()
    {
        return array_map(
            fn ($item) => $this,
            []
        );
    }
}

Current result:

```
--- /home/devlin/Code/XXX/src/Foo.php
+++ /home/devlin/Code/XXX/src/Foo.php
@@ -1,5 +1,12 @@
 <?php
 
 declare(strict_types=1);
 
 namespace App;
@@ -9,7 +16,7 @@
     public function foo()
     {
         return array_map(
-            fn ($item) => $this,
+            static fn ($item) => $this,
             []
         );
     }
```

Expected result:

It should not add the `static` keyword.

### PR DESCRIPTION
**Pull request by drupol**
This PR fix #6439

### CODE REVIEW

**Review by keradus — APPROVED** [review]

**Comment by drupol** [comment]
Thanks!

File: src/Fixer/FunctionNotation/StaticLambdaFixer.php [hunk 1]
```diff
@@ -134,7 +134,7 @@ private function findExpressionEnd(Tokens $tokens, int $index): int
      */
     private function hasPossibleReferenceToThis(Tokens $tokens, int $startIndex, int $endIndex): bool
     {
-        for ($i = $startIndex; $i < $endIndex; ++$i) {
```
Comment (SpacePossum) [inline src/Fixer/FunctionNotation/StaticLambdaFixer.php:137]: maybe this is correct but `private function findExpressionEnd(Tokens $tokens, int $index): int` should return `$nextIndex` and not `$index` ?

Comment (SpacePossum) [reply src/Fixer/FunctionNotation/StaticLambdaFixer.php:137]: nevermind, already merged :D   :+1:

Comment (keradus) [reply src/Fixer/FunctionNotation/StaticLambdaFixer.php:137]: @SpacePossum , if still relevant, do not hesitate to continue on the discussion :)

**Pull request merged**